packages/contracts-bedrock/scripts/go-ffi/go-ffi.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+
+{
+    # Define a lock file
+    LOCKFILE="/tmp/go-ffi.lock"
+
+    # Function to clean up the lock file on exit
+    cleanup() {
+        rm -rf "$LOCKFILE"
+    }
+
+    # Get the directory of the script
+    dir=$(dirname "$(realpath "$0")")
+    # Check if the target binary exists
+    if [[ ! -f "$dir/$1" ]]; then
+        # Wait for the lock file to be released
+        while ! mkdir "$LOCKFILE" > /dev/null; do
+            sleep 0.3s
+        done
```

Comment:
if the lockfile exists, can we just assume that the binary will be built after the lockfile is removed?

Reply:
Then we'll need to store the result of `! mkdir "$LOCKFILE"`, right?